packages/world/src/systems/libraries/TransferLib.sol

```diff
@@ -53,13 +52,38 @@ library TransferLib {
     EntityId chestEntityId = isDeposit ? baseDstEntityId : baseSrcEntityId;
     VoxelCoord memory chestCoord = isDeposit ? dstCoord : srcCoord;
 
+    VoxelCoord memory shardCoord = chestCoord.toLocalEnergyPoolShardCoord();
+    uint128 energyCost = PLAYER_TRANSFER_ENERGY_COST;
+
     EntityId forceFieldEntityId = getForceField(chestCoord);
     uint256 machineEnergyLevel = 0;
     if (forceFieldEntityId.exists()) {
+      energyCost += SMART_CHEST_ENERGY_COST;
       EnergyData memory machineData = updateMachineEnergyLevel(forceFieldEntityId);
       machineEnergyLevel = machineData.energy;
+      require(machineData.energy >= SMART_CHEST_ENERGY_COST, "Machine does not have enough energy");
```

Comment:
feels like some of this stuff (checking enough energy and decreasing from an entity) can be extracted to some util

Reply:
yee I added `transferEnergyFromPlayerToPool` in `EnergyUtils.sol`, curious what else could be moved

Reply:
I think this logic is similar to this for example: https://github.com/tenetxyz/biomes-contracts/pull/35/files#diff-34518cfe1e13072a67175dd059805003eb061c5b3ad4280ce3bc74f0b5cb8e39R85-R90

Reply:
ahh gotcha updated, added a util to `EntityId` `decreaseEnergy`